VideoLocker/src/main/java/org/edx/mobile/task/GetResponsesListTask.java

```diff
@@ -26,8 +26,12 @@ public ThreadComments call() throws Exception {
         try {
             if (threadId != null) {
                 if (isQuestionType) {
+                    if (shouldGetEndorsed) {
+                        return environment.getDiscussionAPI().getResponsesListForQuestion(threadId,
```

Comment:
I don't want to introduce front-end level logic in the API method definitions. Doing it this way preserves flexibility.

I can add a comment here explaining the reason for the separate API calls though, I guess.